diff --git a/boost/serialization/strong_typedef.hpp b/boost/serialization/strong_typedef.hpp
--- a/boost/serialization/strong_typedef.hpp
+++ b/boost/serialization/strong_typedef.hpp
@@ -44,6 +44,7 @@
     operator const T&() const {return t;}                                                                        \
     operator T&() {return t;}                                                                                    \
     bool operator==(const D& rhs) const {return t == rhs.t;}                                                     \
+    bool operator==(const T& lhs) const {return t == lhs;}                                                       \
     bool operator<(const D& rhs) const {return t < rhs.t;}                                                       \
 };
 
diff --git a/tools/build/src/tools/msvc.jam b/tools/build/src/tools/msvc.jam
--- a/tools/build/src/tools/msvc.jam
+++ b/tools/build/src/tools/msvc.jam
@@ -1137,6 +1137,14 @@
         }
         else
         {
+            if [ MATCH "(14.4)" : $(version) ]
+            {
+                if $(.debug-configuration)
+                {
+                    ECHO "notice: [generate-setup-cmd] $(version) is 14.4x" ;
+                }
+                parent = [ path.native [ path.join  $(parent) "..\\..\\..\\..\\..\\Auxiliary\\Build" ] ] ;
+            }
             if [ MATCH "(14.3)" : $(version) ]
             {
                 if $(.debug-configuration)
